lib/smack/SmackRep.cpp

```diff
@@ -207,16 +321,18 @@ const Expr* SmackRep::b2i(const llvm::Value* v) {
 
 const Expr* SmackRep::lit(const llvm::Value* v) {
   using namespace llvm;
-
+  unsigned wd = 0;
   if (const llvm::ConstantInt* ci = llvm::dyn_cast<const llvm::ConstantInt>(v)) {
-    if (ci->getBitWidth() == 1)
+    wd = ci->getBitWidth();
+    wd = TRUNC(wd);
+    if (wd == 1)
       return Expr::lit(!ci->isZero());
-
     uint64_t val = ci->getLimitedValue();
-    if (width > 0 && ci->isNegative())
-      return Expr::fn("$sub", Expr::lit(0, width), Expr::lit(-val, width));
+    if (wd > 0 && ci->isNegative())
+      //return (SmackOptions::BitVectors? Expr::fn(NEG, Expr::lit(val, wd)) : Expr::fn("$sub", Expr::lit(0, width), Expr::lit(-val, width)));
```

Comment:
For this statement in the develop branch, I think there is a potential bug. Variable "val" is actually the two's complement form of the value. So if a short -1 is the literal and passed to this method, then val is 0x000000000000ffff. -val in the Expr::lit(-val, width) will invert all the bits of val and add one which is 0x111111110001 (-65535 in decimal). Then the boogie translation we got is $sub(0, -65535).